packages/ra-input-rich-text/src/RichTextInput.tsx

```diff
@@ -73,117 +73,123 @@ import { TiptapEditorProvider } from './TiptapEditorProvider';
  *     />
  * );
  */
-export const RichTextInput = (props: RichTextInputProps) => {
-    const {
-        className,
-        defaultValue = '',
-        disabled = false,
-        editorOptions = DefaultEditorOptions,
-        fullWidth,
-        helperText,
-        label,
-        readOnly = false,
-        source,
-        sx,
-        toolbar,
-    } = props;
+export const RichTextInput = forwardRef(
+    (props: RichTextInputProps, ref?: React.MutableRefObject<Editor>) => {
+        const {
+            className,
+            defaultValue = '',
+            disabled = false,
+            editorOptions = DefaultEditorOptions,
+            fullWidth,
+            helperText,
+            label,
+            readOnly = false,
+            source,
+            sx,
+            toolbar,
+        } = props;
 
-    const resource = useResourceContext(props);
-    const {
-        id,
-        field,
-        isRequired,
-        fieldState,
-        formState: { isSubmitted },
-    } = useInput({ ...props, source, defaultValue });
+        const resource = useResourceContext(props);
+        const {
+            id,
+            field,
+            isRequired,
+            fieldState,
+            formState: { isSubmitted },
+        } = useInput({ ...props, source, defaultValue });
 
-    const editor = useEditor(
-        {
-            ...editorOptions,
-            editable: !disabled && !readOnly,
-            content: field.value,
-            editorProps: {
-                ...editorOptions?.editorProps,
-                attributes: {
-                    ...editorOptions?.editorProps?.attributes,
-                    id,
+        const editor = useEditor(
+            {
+                ...editorOptions,
+                editable: !disabled && !readOnly,
+                content: field.value,
+                editorProps: {
+                    ...editorOptions?.editorProps,
+                    attributes: {
+                        ...editorOptions?.editorProps?.attributes,
+                        id,
+                    },
                 },
             },
-        },
-        [disabled, editorOptions, readOnly, id]
-    );
+            [disabled, editorOptions, readOnly, id]
+        );
 
-    const { error, invalid, isTouched } = fieldState;
+        const { error, invalid, isTouched } = fieldState;
 
-    useEffect(() => {
-        if (!editor) return;
+        if (ref) {
```

Comment:
You should not set ref during rendering, see https://react.dev/learn/referencing-values-with-refs#best-practices-for-refs

Reply:
done